server-spi/src/main/java/org/keycloak/models/RequiredActionProviderModel.java

```diff
@@ -120,4 +123,12 @@ public Map<String, String> getConfig() {
     public void setConfig(Map<String, String> config) {
         this.config = config;
     }
+
+    public boolean isConfigurable() {
```

Comment:
Do we need `isConfigurable` and `setConfigurable` on the model? It seems that the `configurable` field is not backed in the DB and it is always derived based on the factory (like in `RealmAdapter.entityToModel(RequiredActionProviderEntity entity)`).

Also I believe it is not needed in the JSON representation object, which is used for export/import, but I understand it can be useful for have this in the UI layer. Is it possible to decouple those?

Reply:
Thanks for taking a look! Yes I mostly added the configurable property to the `RequiredActionProviderModel` and `RequiredActionProviderRepresentation` . Initially I thought about deriving the "configurability" of a RequiredAction from a non-empty configuration. However, with this we currently cannot differentiate an empty configuration from a non-configurable required action, as the config propery is currently always initialized with an empty map here: https://github.com/keycloak/keycloak/blob/main/core/src/main/java/org/keycloak/representations/idm/RequiredActionProviderRepresentation.java#L35
If we would change this to be `null` by default, then I could remove the `configurable` from the model and the representation property. 

I was also thinking about using the component model but then realized that required actions already have a database backed config HashMap, so I kept using that.

Reply:
I understand that the info about "configurable" is needed in the UI. But I think it is not needed in the model layer and might be cleaner to remove it from here.

I see the options like:

1) Keep `configurable` at the model layer and `RequiredActionProviderRepresentation`. Not so clean IMO.
2) Use the `null` to indicate. It can work, but just afraid a bit of things like `NullPointerException` etc.
3) Add stuff inside admin UI java extension (module `rest/admin-ui-ext` in the workspace) and make admin console to use it. I can imagine something like this in the package `org.keycloak.admin.ui.rest.model` inside admin UI extension (the stuff in the module `rest`, which contains REST endpoints used solely by admin console):

```
public class ConfigurableRequiredActionProviderRepresentation extends RequiredActionProviderRepresentation {

    private boolean configurable;

    // Getter and setter for "configurable" here ...
}
```

And then make `org.keycloak.admin.ui.rest.AuthenticationManagementResource` to return required actions like those, so the admin console UI is able to figure if action is configurable or not. The disadvantage of the extension is additional code.

My vote is either (2) or (3). WDYT?